### PR TITLE
Add the validator for --build-env

### PR DESCRIPTION
**Pull request by smile37773**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repository and upgrade the version in the pull request but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
spring

File: src/spring/azext_spring/_validators_enterprise.py [hunk 1]
```diff
@@ -43,7 +43,17 @@ def validate_build_env(cmd, namespace):
         if isinstance(namespace.build_env, list):
             env_dict = {}
             for item in namespace.build_env:
-                env_dict.update(validate_tag(item))
+                if item:
+                    comps = item.split('=', 1)
+                    if len(comps) <= 1:
+                        raise ArgumentUsageError("The value of env {} should not be empty.".format(item))
+                    else:
+                        if match(r"^[-._a-zA-Z][-._a-zA-Z0-9]*$", comps[0]):
```
Comment (zhoxing-ms) [inline src/spring/azext_spring/_validators_enterprise.py:51]: Can you add some comments and examples to explain the correct format that this regular expression can match

File: src/spring/azext_spring/_validators_enterprise.py [hunk 2]
```diff
@@ -43,7 +43,17 @@ def validate_build_env(cmd, namespace):
...
+                            result = {}
+                            result = {comps[0]: comps[1]}
+                            env_dict.update(result)
+                        else:
+                            raise ArgumentUsageError("The env name {} is not allowed, the valid env name should follow the pattern '[-._a-zA-Z][-._a-zA-Z0-9]*'.".format(comps[0]))
```
Comment (zhoxing-ms) [inline src/spring/azext_spring/_validators_enterprise.py:56]: Same as above, it is recommended to add an example to the error message to enhance its readability

File: src/spring/azext_spring/_validators_enterprise.py (hunk 1 above)
Comment (smile37773) [reply src/spring/azext_spring/_validators_enterprise.py:51]: OK, add examples in the help message.

File: src/spring/azext_spring/_validators_enterprise.py (hunk 2 above)
Comment (smile37773) [reply src/spring/azext_spring/_validators_enterprise.py]: OK.

File: src/spring/azext_spring/_help.py [hunk 3]
```diff
@@ -257,6 +257,8 @@
       text: az spring app deploy -n MyApp -s MyCluster -g MyResourceGroup --container-image contoso/your-app:v1 --container-registry myacr.azurecr.io --registry-username <username> --registry-password <password>
     - name: Deploy with Application Configuration Service config file patterns to an app.
       text: az spring app deploy -n MyApp -s MyCluster -g MyResourceGroup --config-file-patterns MyPatterns --artifact-path app.jar
+    - name: Deploy a pre-built jar to an app with build env(For Enterprise tier only).
```
Comment (zhoxing-ms) [inline src/spring/azext_spring/_help.py:260]:
```suggestion
    - name: Deploy a pre-built jar to an app with build env (For Enterprise tier only).
```

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**